go/vt/vtgate/executor_framework_test.go

```diff
@@ -572,10 +571,47 @@ func executorStream(executor *Executor, sql string) (qr *sqltypes.Result, err er
 	return qr, nil
 }
 
-func testQueries(t *testing.T, sbcName string, sbc *sandboxconn.SandboxConn, wantQueries []*querypb.BoundQuery) {
+func testQueries(t *testing.T, sbc *sandboxconn.SandboxConn, wantQueries []*querypb.BoundQuery) {
 	t.Helper()
-	if !reflect.DeepEqual(sbc.Queries, wantQueries) {
-		t.Errorf("%s.Queries:\n%+v, want\n%+v\n", sbcName, sbc.Queries, wantQueries)
+	idx := 0
+	for _, query := range sbc.Queries {
+		if strings.HasPrefix(query.Sql, "savepoint") || strings.HasPrefix(query.Sql, "rollback to") {
+			continue
+		}
+		if len(wantQueries) < idx {
+			t.Errorf("got more queries than expected")
+		}
+		require.Equal(t, query.BindVariables, wantQueries[idx].BindVariables)
+		got := query.Sql
+		expected := wantQueries[idx].Sql
+		idx++
+		assert.Equal(t, expected, got)
+	}
+}
+
+func testQueriesWithSavepoint(t *testing.T, sbc *sandboxconn.SandboxConn, wantQueries []*querypb.BoundQuery) {
```

Comment:
same this as above - since this method is just asserting state, could we name it something with assert, check or require?

Reply:
I am thinking we could probably move these two methods in the `sandboxconn` package and make them public so we can reuse them everywhere.

Reply:
they are used only for executor tests, if it is used/required anywhere else we can move it later.